components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/R71498PhenoTips2155DataMigration.java

```diff
@@ -132,264 +202,638 @@ public Object doInHibernate(Session hSession) throws HibernateException, XWikiEx
             Query q =
                 this.session.createQuery("select distinct o.name from BaseObject o, BaseObject ro where"
                     + " o.className = :patclass and o.name <> 'PhenoTips.PatientTemplate'"
-                    + " and ro.name = o.name and ro.className = :relclass");
```

Comment:
Because I need to get all patients, including those without family studies data.

To be precise, I need to get all patients that may be referred TO by a family studies relation, but was not sure how to quickly do the query which gets all patients that either have a relationship, or are referred to by a relationship. I assumed this part will be fast enough and only done once per instance, so not a big deal to optimize the query to get only those patients I need.